core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -93,6 +93,19 @@ class SparkEnv (
     // actorSystem.awaitTermination()
 
     // Note that blockTransferService is stopped by BlockManager since it is started by it.
+    
+    // If we only stop sc, but the driver process still run as a services then we need to delete
+    // the tmp dir, if not, it will create too many tmp dirs.
+    // We only need to delete the tmp dir create by driver, because sparkFilesDir is point to the
+    // current working dir in executor which we do not need to delete.
+    if (SparkContext.DRIVER_IDENTIFIER == executorId) {
```

Comment:
@srowen sorry, I don't very clear. You mean we can not use the executorId to distinguish the driver and executor?